src/routes/transactions/entities/swap-order-info.entity.ts

```diff
@@ -7,168 +7,132 @@ import {
   ApiProperty,
   ApiPropertyOptional,
 } from '@nestjs/swagger';
+import { OrderStatus } from '@/domain/swaps/entities/order.entity';
 
 export class TokenInfo {
+  @ApiProperty({ description: 'The token address' })
+  address: `0x${string}`;
+
+  @ApiProperty({ description: 'The token decimals' })
+  decimals: number;
+
   @ApiPropertyOptional({
     type: String,
     nullable: true,
-    description: 'The token logo',
+    description: 'The logo URI for the token',
   })
-  logo: string | null;
+  logoUri: string | null;
+
+  @ApiProperty({ description: 'The token name' })
+  name: string;
 
   @ApiProperty({ description: 'The token symbol' })
   symbol: string;
 
-  @ApiProperty({
-    description: 'The token amount in decimal format',
-  })
-  amount: string;
+  @ApiProperty({ description: 'The token trusted status' })
+  trusted: boolean;
 
-  constructor(args: { logo: string | null; symbol: string; amount: string }) {
-    this.logo = args.logo;
+  constructor(args: {
+    address: `0x${string}`;
+    decimals: number;
+    logoUri: string | null;
+    name: string;
+    symbol: string;
+    trusted: boolean;
+  }) {
+    this.address = args.address;
+    this.decimals = args.decimals;
+    this.logoUri = args.logoUri;
+    this.name = args.name;
     this.symbol = args.symbol;
-    this.amount = args.amount;
+    this.trusted = args.trusted;
   }
 }
 
 @ApiExtraModels(TokenInfo)
-export abstract class SwapOrderTransactionInfo extends TransactionInfo {
+export class SwapOrderTransactionInfo extends TransactionInfo {
   @ApiProperty({ enum: [TransactionInfoType.SwapOrder] })
-  override type: TransactionInfoType.SwapOrder;
+  override type = TransactionInfoType.SwapOrder;
 
   @ApiProperty({ description: 'The order UID' })
-  orderUid: string;
+  uid: string;
 
   @ApiProperty({
-    enum: ['open', 'fulfilled', 'cancelled', 'expired', 'presignaturePending'],
+    enum: OrderStatus,
   })
-  status:
-    | 'open'
-    | 'fulfilled'
-    | 'cancelled'
-    | 'expired'
-    | 'presignaturePending';
+  status: OrderStatus;
 
   @ApiProperty({ enum: ['buy', 'sell'] })
-  orderKind: 'buy' | 'sell';
-
-  @ApiProperty({ description: 'The sell token of the order' })
-  sellToken: TokenInfo;
-
-  @ApiProperty({ description: 'The buy token of the order' })
-  buyToken: TokenInfo;
+  kind: 'buy' | 'sell';
 
   @ApiProperty({ description: 'The timestamp when the order expires' })
-  expiresTimestamp: number;
+  validTo: number;
```

Comment:
Nit: I'd tweak this to `validUntil` or match that of the comment, e.g. `expiresAt`.

Reply:
I've renamed it to be closer to the `Order` object. I do prefer `validUntil` though.

Reply:
Adjust in [af164ec](https://github.com/safe-global/safe-client-gateway/pull/1426/commits/af164ec085dc60df6de5c9620f3024f04b4a9adb)